### PR TITLE
feat: Stop advancing thread on shutdown

### PR DESCRIPTION
**Pull request by oblique**

### CODE REVIEW

File: client/src/client.rs [hunk 1]
```diff
@@ -14,7 +14,7 @@ use config::{CheckpointFallback, Config};
 use consensus::{types::Header, ConsensusClient};
 use execution::types::{CallOpts, ExecutionBlock};
 use log::{error, info, warn};
-use tokio::sync::RwLock;
+use tokio::sync::{Mutex, RwLock};
```
Comment (zvolin) [inline client/src/client.rs]: As there is no need to persist the mutex across awaits I think we can use lighter `std::sync::Mutex`

File: client/src/client.rs [hunk 2]
```diff
@@ -476,7 +487,33 @@ impl<DB: Database> Client<DB> {
         };
     }
 
+    pub async fn get_last_checkpoint(&self) -> Option<String> {
```
Comment (zvolin) [inline client/src/client.rs]: this doesn't seem related, do you want to include it?

Comment (oblique) [reply client/src/client.rs]: I use it in the ethereum-canister. Do you prefer it in a separate PR?

Comment (zvolin) [reply client/src/client.rs]: not a big deal for me but I'd prefer separate PR, this feels a bit sneaky

**Review by zvolin — APPROVED** [review]
